sandy/njoy.py

```diff
@@ -691,6 +691,59 @@ def _errorr_input(endfin, pendfin, errorrout, mat,
     return "\n".join(text) + "\n"
 
 
+def _groupr_input(endfin, pendfin, pendfout, mat,
+                  ign=2, igg=0, iwt=2, lord=0, nsigz=1,
+                  temp=NJOY_TEMPERATURES,
+                  iprint=False,
+                  **kwargs):
+    """
+    Write GROUPR input
+
+    Parameters
+    ----------
+    endfin : `int`
+        tape number for input ENDF-6 file
+    pendfin : `int`
+        tape number for input PENDF file
+    pendfout : `int`
+        tape number for output PENDF file
+    mat : `int`
+        MAT number
+    ign : `int`, optional
+        neutron group option (default is 2, csewg 239-group structure)
+    igg : `int`, optional
+        gamma group option. (default is 0, no structure)
+    iwt : `int`, optional
+        weight function option (default is 2, constant)
+    lord : TYPE, optional
+        Legendre order. The default is 0.
+    nsigz : TYPE, optional
+        Number of sigma zeroes. The default is 1.
+    temp : terable of `float`
+        iterable of temperature values in K (default is 293.6 K)
+    iprint : `bool`, optional
+        print option (default is `False`)
+
+    Returns
+    -------
+    `str`
+        groupr input text
+
+    """
+    text = ["groupr"]
+    text += [f"{endfin:d} {pendfin:d} 0 {pendfout:d} /"]
+    ntemp = len(temp)
+    printflag = int(iprint)
+    text += [f"{mat:d} {ign:d} {igg:d} {iwt:d} {lord:d} {ntemp:d} {nsigz:d} {printflag:d} /"]
+    text += ["/"]
+    text += [" ".join(map("{:.1f}".format, temp)) + "/"]
+    text += ["1e+10/"]
```

Comment:
Here you give only one sigz value. The number of values is indicated by `nsigz`.
If the user gives `nsigz=2`, then NJOY will fail.

We have two options:
- we set `nsigz=1` and we remove it from the input arguments (simple implementation, no flexibility)
- we use a variable `sigz` in input, as we do for `temp`. Then `nsigz` will be derived, as for `ntemp`